products/azure_rm_batchaccount/api.yaml

```
@@ -0,0 +1,101 @@
+        required: true
+        azure_sdk_references: ['accountName']
+      - !ruby/object:Api::Azure::Type::str
+        name: 'location'
+        description: 'The region in which to create the account.'
+        required: true
+        azure_sdk_references: ['/location']
+      - !ruby/object:Api::Azure::Type::dict
+        name: 'tags'
+        description: 'The user-specified tags associated with the account.'
+        required: true
+        azure_sdk_references: ['/tags']
+      - !ruby/object:Api::Azure::Type::dict
+        name: 'properties'
+        description: 'The properties of the Batch account.'
+        required: true
+        azure_sdk_references: ['/properties']
+        properties:
+          - !ruby/object:Api::Azure::Type::dict
+            name: 'auto_storage'
+            description: 'The properties related to the auto-storage account.'
+            required: true
+            azure_sdk_references: ['/properties/autoStorage']
+            properties:
+              - !ruby/object:Api::Azure::Type::str
+                name: 'storage_account_id'
+                description: 'The resource ID of the storage account to be used for auto-storage account.'
+                required: true
+                azure_sdk_references: ['/properties/autoStorage/storageAccountId']
+          - !ruby/object:Api::Azure::Type::dict
+            name: 'pool_allocation_mode'
+            description: 'The pool allocation mode also affects how clients may authenticate to the Batch Service API. If the mode is BatchService, clients may authenticate using access keys or Azure Active Directory. If the mode is UserSubscription, clients must use Azure Active Directory. The default is BatchService.'
+            required: true
+            azure_sdk_references: ['/properties/poolAllocationMode']
+          - !ruby/object:Api::Azure::Type::dict
+            name: 'key_vault_reference'
+            description: 'A reference to the Azure key vault associated with the Batch account.'
+            required: true
+            azure_sdk_references: ['/properties/keyVaultReference']
+            properties:
+              - !ruby/object:Api::Azure::Type::str
+                name: 'id'
+                description: 'The resource ID of the Azure key vault associated with the Batch account.'
+                required: true
+                azure_sdk_references: ['/properties/keyVaultReference/id']
+              - !ruby/object:Api::Azure::Type::str
+                name: 'url'
+                description: 'The URL of the Azure key vault associated with the Batch account.'
+                required: true

```

This property can not be updated.

```suggestion
                required: true
                input: true
```